spec/integration/decoding_payloads_spec.rb

```diff
@@ -0,0 +1,48 @@
+class TwitterSubscriber < ActionSubscriber::Base
+  def tweet
+    $messages << {
+      :decoded => payload,
+      :raw => raw_payload,
+    }
+  end
+end
+
+
+describe "Payload Decoding", :integration => true do
+  let(:connection) { subscriber.connection }
+  let(:subscriber) { TwitterSubscriber }
+  let(:json_string) { '{"foo": "bar"}' }
+
+  it "decodes json by default" do
+    ::ActionSubscriber.auto_subscribe!
+    channel = connection.create_channel
+    exchange = channel.topic("events")
+    exchange.publish(json_string, :routing_key => "twitter.tweet", :content_type => "application/json")
+    sleep 0.1
+
+    expect($messages).to eq Set.new([{
+      :decoded => JSON.parse(json_string),
+      :raw => json_string,
+    }])
+  end
+
+  context "Custom Decoder" do
+    let(:content_type) { "foo/type" }
+
+    before { ::ActionSubscriber.config.add_decoder(content_type => lambda{ |payload| :foo }) }
```

Comment:
is this clearer? It sort of provides isolation, but hopefully once I start refactoring the internals we can get this isolation without having to clobber global state